src/error.ts

```diff
@@ -744,20 +744,20 @@ function isRecoveringError(err: MongoError) {
     return SDAM_RECOVERING_CODES.has(err.code);
   }
 
-  return /not master or secondary/.test(err.message) || /node is recovering/.test(err.message);
+  return /not primary or secondary/.test(err.message) || /node is recovering/.test(err.message);
 }
 
-function isNotMasterError(err: MongoError) {
+function isNotPrimaryError(err: MongoError) {
   if (typeof err.code === 'number') {
     // If any error code exists, we ignore the error.message
-    return SDAM_NOTMASTER_CODES.has(err.code);
+    return SDAM_NOTPRIMARY_CODES.has(err.code);
   }
 
   if (isRecoveringError(err)) {
     return false;
   }
 
-  return /not master/.test(err.message);
+  return /not primary/.test(err.message);
```

Comment:
I would use the constant mentioned in the other comment here to test for `(not primary|not master)`

Reply:
I would also add unit tests to test `isSDAMUnrecoverableError` in test/unit/error.test.js for both cases in the error message.

Reply:
I've added unit tests